yb-voyager/src/queryparser/unsupported_constructs.go

```diff
@@ -0,0 +1,110 @@
+package queryparser
+
+import (
+	"fmt"
+
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+const (
+	ADVISORY_LOCKS = "Advisory Locks"
+	SYSTEM_COLUMNS = "System Columns"
+	XML_FUNCTIONS  = "XML Functions"
+)
+
+// To Add a new unsupported query construct implement this interface
+type UnsupportedConstructDetector interface {
```

Comment:
I like your general idea of structuring it as detectors! But, do we really need an interface with structs implementing them? 
How about we just work with functions? https://go.dev/play/p/zNiENQOAmVI

Reasoning is same as in my earlier PR. None of these detectors really need to maintain any state over the lifetime of the struct; it just performs some operation, and returns some values. A function is better-suited. For example, look at the `YBValueConverterSuite`

Reply:
Yeah, i agree generally if we don't need to maintain any state we should avoid interface/structs

But couple of points here to choose interface:
1. Current design enforces the implementation of Detect() for every new detector.
2. Due to struct, in future if we requirements like, for example
- tell how many times xmlparse()/xpath() function has occured
- any niche check for a particular detector
- or any other specific stats based on the workload queries

overall i think interface approach is less simpler than function based, but in future for this codepath as we enhance this might become more useful and easier to implement complex checks.

Reply:
>Current design enforces the implementation of Detect() for every new detector.

This will be true even if we use funcs because every detector will have to be defined to meet that same function signature. 

>Due to struct, in future if we requirements like, for example
tell how many times xmlparse()/xpath() function has occured
any niche check for a particular detector
or any other specific stats based on the workload queries

Since your Detect() is defined to operate on the `protoreflect.Message` , things like stats can be computed in your processor function. 

However, there is one particular case that comes to mind: What if the detection logic requires to look at multiple nodes? In that case you can't just detect just by looking at a single message, you might only be able to detect after parsing the entire tree.  In which case I would define the interface like this instead: 

```
type UnsupportedConstructDetector interface {
    ProcessMessage(msg protoreflect.Message) (error)
    GetUnsupportedConstructs() ([]string)
```
Does that make sense? We can also discuss this on call if it's easier.

Reply:
Thinking out loud 
> What if the detection logic requires to look at multiple nodes?
Even in this case, their will be parent node(may be root node) on which we need to look for 2 nodes under it. Which can be done with the current approach also.